vpr/src/base/vpr_types.h

```diff
@@ -1344,15 +1344,21 @@ struct t_det_routing_arch {
     std::string write_rr_graph_filename;
 };
 
+/*
+ * INC_DIRECTION: wire driver is positioned at the low-coordinate end of the wire.
+ * DEC_DIRECTION: wire_driver is positioned at the high-coordinate end of the wire.
+ * BI_DIRECTION: wire has multiple drivers, so signals can travel either way along the wire
+ * INVALID_DIRECTION: wire does not have a valid direction set
+ */
 enum e_direction : unsigned char {
     INC_DIRECTION = 0,
     DEC_DIRECTION = 1,
     BI_DIRECTION = 2,
-    NO_DIRECTION = 3,
+    INVALID_DIRECTION = 3,
```

Comment:
Actually, the ``NUM_DIRECTION`` represents the invalid direction. It was made to be last one in the enumeration and considered as the invalid value.
We can consider to remove this one and use ``NUM_DIRECTIONS`` as the invalid value to return. This can reduce the size this enumeration and follow the same convention as other enumerations.

Reply:
@vaughnbetz do you have any input on this? The code currently has `INVALID_DIRECTION` as the fourth enumeration. I can change that back to `NO_DIRECTION`, or even remove it entirely. (I think that is what Xifan is meaning in this comment)

Reply:
OK with me to change back to NO_DIRECTION (or now, Direction::NONE).  Should comment how that happens / usage though: I believe it is for nodes where direction is not meaningful, such as input or output pins (not wires).